### PR TITLE
fix native crash in DynamicRealmObject#setList()

### LINKED ISSUE #3549: DynamicRealmObject#setList() causes native crash if the specified field is not a List
Calling `DynamicRealmObject#setList()` against an object from `DynamicRealm` causes a native crash if the field type is not a `List`.

### PR DESCRIPTION
**Pull request by zaki50**
fixes #3549

This PR requires #3551 to be merged.

### CODE REVIEW

**Comment by zaki50** [comment]
merged #3551 into this branch.

Ready for review @realm/java

File: realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java [hunk 1]
```diff
@@ -246,8 +266,12 @@ public double getDouble(String fieldName) {
      */
     public String getString(String fieldName) {
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java]: Isnt't there some types missing like Date/Object ?

**Comment by zaki50** [comment]
@nhachicha I think that line should be:

```
if (columnType == RealmFieldType.INTEGER || columnType == RealmFieldType.OBJECT) {
```

**Comment by zaki50** [comment]
retest this please

**Comment by zaki50** [comment]
I'll merge this if Mr. Jenkins says :+1:

**Comment by zaki50** [comment]
retest this please

**Pull request merged**